src/simulator/ackermann_model.cpp

```diff
@@ -19,6 +20,7 @@ CONFIG_STRING(cDriveTopic, "drive_callback_topic");
 
 
 AckermannModel::AckermannModel(const string& config_file, ros::NodeHandle n) :
+    EntityBase(),
     angular_error_(0, 1) {
   config_reader::ConfigReader({config_file});
```

Comment:
You should add a `config_reader::ConfigReader` object as a member variable to the class; right now, [the lifetime of the unnamed object is the lifetime of the automatic variable scope, which is to the until the `;`](https://stackoverflow.com/questions/2298781/why-do-un-named-c-objects-destruct-before-the-scope-block-ends), which means the destructor will be called on the very next line, [meaning that the variables will not live update if the config file is re-saved while running.](https://github.com/ut-amrl/config-reader/blob/master/include/config_reader/config_reader.h#L180) Of course, changing the topic name won't work without a rerun, but things like parameter tweaking should be able to be done just by editing the file without restarting the simulator.